### DISCUSSION TITLE
How do I use bash completions?

### DISCUSSION THREAD

**Discussion by anihm136** [z-shell/zi#213] [Q&A] [answered]
Hi all, new Zi user here. I have some plugins/commands which do not have native zsh completions, and instead rely (in some form) on the bash completion compatibility of zsh (using `bashcompinit`). Since this relies on `compdef`, which Zi handles opaquely, I have been unable to get these completions to work by using `autoload -Uz bashcompinit && bashcompinit` in my config. Is there any documentation/guidance on how to work with this? Apologies if I have missed something.

**Comment by anihm136** [comment, marked as answer]
Just like always, I found the answer right after posting the question. For reference, my solution was to put all of my bash completion-related stuff in a single file and load it as a snippet. Then zi intercepts the compdef calls and handles it correctly with cdreplay. I'm marking this as solved for now

**Reply by ss-o** [reply]
Hey :wave: 

That was quick :+1: I suggest generating completions with tools like https://github.com/RobSis/zsh-completion-generator. :dart:

**Reply by anihm136** [reply]
First I'm hearing of this, thanks! Will give it a spin soon
